gocourse17/server/server.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"github.com/vikibell/prjctr-golang-beginning/gocourse17/service"
+	"google.golang.org/grpc"
+	"log"
+	"net"
+
+	pb "github.com/vikibell/prjctr-golang-beginning/gocourse17/grpcapi"
+)
+
+type server struct {
+	pb.UnimplementedReviewServer
+
+	history service.ReviewHistory
+}
+
+func (s *server) GetHistory(_ context.Context, request *pb.HistoryRequest) (*pb.HistoryResponse, error) {
+	driverId := request.GetDriverId()
+	if driverId <= 0 {
+		return &pb.HistoryResponse{}, errors.New("invalid driver id")
```

Comment:
Можна просто `nil` замість `&pb.HistoryResponse{}`.
```suggestion
		return nil, errors.New("invalid driver id")
```

Reply:
Замість `errors.New("invalid driver id")` пропоную використати gRPC статус коди.

Почитати тут:

- https://grpc.io/docs/guides/error/
- https://github.com/grpc/grpc-go/tree/master/examples/features/error_handling
- https://github.com/grpc/grpc-go/tree/master/examples/features/error_details